step-release-vis/src/app/components/environment/environment_test.ts

```diff
@@ -161,4 +162,45 @@ describe('EnvironmentComponent', () => {
       expect(candName).toEqual(environmentServiceStub.candName);
     });
   });
+
+  describe('computeSnapshotIntervals', () => {
+    beforeEach(() => {
+      component.startTimestamp = 1;
+    });
+
+    it('no timestamps', () => {
+      component.displayedSnapshots = [];
+      component.computeSnapshotIntervals();
+
+      expect(component.snapshotIntervals.length).toEqual(0);
+    });
+
+    it('only one timestamp', () => {
+      component.displayedSnapshots = [{timestamp: 1, candidatesList: []}];
+      component.computeSnapshotIntervals();
+
+      const expectedResult: SnapshotInterval[] = [
+        new SnapshotInterval(1, 1, component.displayedSnapshots[0]),
```

Comment:
why 1,1 do we have a single pixel with? I would expect 0, width here, if I got it right what you do